examples/error-type-binding-pattern/error_type_binding_pattern.bal

```diff
@@ -1,37 +1,38 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+    // The values in the detail mapping will be set to new variables `info` and `fatal`.
+    var error(reason, info = info, fatal = fatal) = getSampleError();
     io:println("Reason String: " + reason);
-    io:println(io:sprintf("Detail Mapping: %s", detail));
-
-    // The detail-mapping value can further be destructured using a record-binding pattern.
-    SampleError error(reasonTwo, { detail: detailTwo, fatal }) = getSampleError();
-    io:println("Reason String: " + reasonTwo);
-    io:println(io:sprintf("Detail Mapping Field One: %s", detailTwo));
-    io:println(io:sprintf("Detail Mapping Field Two: %s", fatal));
-
-    // Error-type binding patterns can be used with `var` to infer the type from the right hand side.
-    // Since the types of the new variables are based on the type of the type-binding pattern, using `var` will
-    // infer the types from the right hand side.
-    var error(vReason, vDetail) = getSampleError();
-    // Type of `vReason` is inferred as `string`.
-    io:println("Reason String: " + vReason);
-    // Type of `vDetail` is inferred as `map<anydata|error>`.
-    io:println(io:sprintf("Detail Mapping: %s", vDetail));
+    io:println("Info: ", info);
+    io:println("Fatal: ", fatal);
+
+    // The detail mapping can also be destructured using a rest parameter.
+    // `params` will be of type `map<anydata|error>`, and will have the 
```

Comment:
```suggestion
    // `params` will be of the type `map<anydata|error>` and will have the 
```

Reply:
Error object has two fields called "reason" and "detail". The "detail" result (mapping) is stated here.